Kernel/makeall.sh

```diff
@@ -1,6 +1,10 @@
 #!/bin/bash
 set -e
 
+# Get user and group details for setting qemu disk image ownership
```

Comment:
It would be even nicer to put these values into variables so we don't have to worry about cleaning up the files.
```bash
build_user=$(id -u)
build_group=$(id -g)
```

And then above, just
```bash
chown $build_user:$build_group ...
```
```

Reply:
I had to write the user and group out to a file because both user and group get changed to root first thing in makeall.sh currently, as it instantly switches to sudo.

Reply:
When does makeall actually need sudo?

Reply:
As soon as makeall.sh runs `sudo id` the user and group get set to root for everything after that point.

Reply:
Wait, that doesn't make any sense, sudo id should be a self-contained execution of id.
If you run id -u on the next line, does it really print out "0" on your system?

The only thing makeall.sh needs sudo for is invoking sudo ./sync.sh

Reply:
The reason it does `sudo id` early is to force the `sudo` password prompt to show up immediately if you're not in sudo's cached-authentication mode.

Reply:
I have changed my patch to export the user and group as environment variables instead.

Reply:
Seems I don't know how to use github yet. Do I need to submit a new pull request now for my updated commit?

Reply:
No no, you're all good :)

Reply:
Yip!